src/gui/qgsmapcanvas.h

```diff
@@ -548,6 +570,13 @@ class GUI_EXPORT QgsMapCanvas : public QGraphicsView
     //! @note added in 2.12
     void layerStyleOverridesChanged();
 
+    /**
+     * Emitted when the map theme set for the canvas is changed.
```

Comment:
From the comment it is not entirely clear when is the signal is emitted: whether on map canvas assignment to a different theme or when the currently assigned theme gets modified somehow...